src/transform/src/reduction.rs

```diff
@@ -215,10 +215,8 @@ impl FoldConstants {
                             let mut next_rows = Vec::new();
                             for (old_row, old_count) in old_rows {
                                 for (new_row, new_count) in rows.iter() {
-                                    let old_datums = old_row.unpack();
-                                    let new_datums = new_row.unpack();
                                     next_rows.push((
-                                        row_packer.pack(old_datums.iter().chain(new_datums.iter())),
+                                        row_packer.pack(old_row.iter().chain(new_row.iter())),
                                         old_count * *new_count,
```

Comment:
I haven't read all of this code, the local change seems totally fine but I was a little surprised at the multiply

Reply:
wait, just meaning it looks funny, or surprised that the code multiplies?

Reply:
surprised that it multiplies because it looks as if the two rows are being smushed together (would have expected to see addition i guess) ofc this is probably a total misread.

Reply:
Ah, but it is in `Join`. So if i have `(a, 20)` and `(b, 23)` I want as output `((a, b), 460)`.